src/lib/modules/ProofModule.ts

```diff
@@ -0,0 +1,36 @@
+import { ConnectionRecord } from '../storage/ConnectionRecord';
+import { createOutboundMessage } from '../protocols/helpers';
+import { MessageSender } from '../agent/MessageSender';
+import { ProofService, ProofRequestTemplate } from '../protocols/proof/ProofService';
+import { ProofRecord } from '../storage/ProofRecord';
+/**
+ * The fuctionalities of this Class is used to Send proof request
+ */
+export class ProofModule {
+  private proofService: ProofService;
+  private messageSender: MessageSender;
+
+  public constructor(proofService: ProofService, messageSender: MessageSender) {
+    this.proofService = proofService;
+    this.messageSender = messageSender;
+  }
+
+  /**
+   * This method is used to send proof request
+   * @param connection : Connection to which issuer wants to issue a credential
+   * @param ProofRequestTemplate : Template used to send proof request
+   */
+  public async sendProofRequest(connection: ConnectionRecord, proofRequestTemplate: ProofRequestTemplate) {
+    const proofOfferMessage = await this.proofService.createProofRequest(connection, proofRequestTemplate);
+    const outboundMessage = createOutboundMessage(connection, proofOfferMessage);
+    await this.messageSender.sendMessage(outboundMessage);
+  }
+
+  public async getProofs(): Promise<ProofRecord[]> {
+    return this.proofService.getAll();
+  }
+
+  public async find(id: string) {
+    return this.proofService.find(id);
+  }
+}
```

Comment:
I would try to add this functionality into `CredentialsModule`. It's kind of related and I would like to simplify API with modules working as facades. If we see it's really bloated with a lot of unrelated stuff we can separate it later

Reply:
Of course, we would need to update `find` methods to `findCredential` and `findProof`

Reply:
Ok, as I went away from the keyboard for a few minutes I start thinking and realize it's maybe not bad to have separate module for this functionality. So let's keep it as it is.